pkg/autoscaler/autoscaler.go

```diff
@@ -82,7 +83,13 @@ func New(
 func (a *Autoscaler) Update(deciderSpec DeciderSpec) error {
 	a.specMux.Lock()
 	defer a.specMux.Unlock()
+
+	a.counterMux.Lock()
+	defer a.counterMux.Unlock()
+
 	a.deciderSpec = deciderSpec
+	a.podCounter.UpdateName(deciderSpec.ServiceName)
```

Comment:
I am pretty sure this is immutable.

Reply:
Do you mean the service name? Because the decider spec swap is total, it's not merged. I noticed it in the `autoscaler_test`: https://github.com/jchesterpivotal/serving/blob/add-current-replicas-to-scale-method/pkg/autoscaler/autoscaler_test.go#L201

Which is how I started working on how to handle `Update`.

Reply:
Yeah in real life it's immutable. Test has to change rather.

Reply:
Cool. In that case I can dump everything related to that, which should simplify things.